src/context/network.js

```diff
@@ -138,8 +149,22 @@ function networkReducer(state, action) {
 
         // FIXME: we need to keep the old ID. Perhaps we should consider how we are handled the IDs.
         return {
-            ...state, interfaces:
-            { ...interfaces, [oldIface.id]: { ...oldIface, ...action.payload, id: oldIface.id } }
+            ...state,
+            interfaces: {
+                ...interfaces, [oldIface.id]: { ...oldIface, ...action.payload, id: oldIface.id }
```

Comment:
Missing `pending: false` here. I switched the interface state and the spinner never gone because of `pending: true`.

And probably we should _reset_ it in the `CONNECTION_ERROR` action too.